[docs][data grid] Copyedit the Aggregation doc (#16200)
mui/mui-x: docs/data/data-grid/aggregation/aggregation.md

# Data Grid - Aggregation [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

<p class="description">Add aggregation functions to the Data Grid to let users combine row values.</p>

The Data Grid Premium provides tools to give end users the ability to aggregate and compare row values.

> It includes [built-in functions](#built-in-functions) to cover common use cases such as sum, average, minimum, and maximum, as well as the means to [create custom functions](#creating-custom-functions) for all other needs.

Reviewer: not sure if there should be a `,` here

```suggestion
It includes [built-in functions](#built-in-functions) to cover common use cases such as sum, average, minimum and maximum, as well as the means to [create custom functions](#creating-custom-functions) for all other needs.
```

Reviewer 2: That's called an Oxford (or serial) comma, and I'm a big fan of it. 😁 It seems like a very minor detail but English can be a pretty ambiguous language so I always prefer to err on the side of clarity. In one case, the lack of an Oxford comma in an employment contract led to a successful [multi-million-dollar lawsuit](https://archive.ph/Ggz0x)!

End users can aggregate rows through the Data Grid interface by opening the column menu and selecting from the items under **Aggregation**.
The aggregated values are rendered in a footer row at the bottom of the Grid.

{{"demo": "AggregationInitialState.js", "bg": "inline", "defaultCodeOpen": false}}

:::info
This document covers client-side implementation.

> For aggregation on the server side, see [Server-side data—Aggregation](/x/react-data-grid/server-side-data/aggregation/).

Reviewer: The title for that page in the docs is different:
```suggestion
For aggregation on the server side, see [Server-side aggregation](/x/react-data-grid/server-side-data/aggregation/).
```

{{"demo": "AggregationDisabled.js", "bg": "inline", "defaultCodeOpen": false}}

### For specific columns

To disable aggregation on a specific column, set the `aggregable` property on its column definition (`GridColDef`) to `false`.

In the example below, the **Year** column is not aggregable since its `aggregable` property is set to `false`.

{{"demo": "AggregationColDefAggregable.js", "bg": "inline", "defaultCodeOpen": false}}

## Aggregating non-aggregable columns

To apply aggregation programmatically on non-aggregable columns (columns with `aggregable: false` in the [column definition](/x/api/data-grid/grid-col-def/)), you can provide the aggregation model in one of the following ways:

- Pass `aggregation.model` to the `initialState` prop. This initializes aggregation with the provided model.
- Provide the `aggregationModel` prop. This controls aggregation with the provided model.

> - Call the API method `setAggregationModel()`. This sets the aggregation with the provided model.

Reviewer: sets aggregation sounds a bit weird
could we maybe use a different term?

Suggestion:
```suggestion
- Call the API method `setAggregationModel()`. This applies an aggregation with the provided model.
```
